### ISSUE TITLE
Unresolved damage/energy/etc on movement is not getting resolved until following turn

### ISSUE THREAD

**Issue by Zheneq** [Zheneq/hc#36]
So the actor gets revived and carries unresolved damage into the next turn

**Comment by Zheneq** [comment]
Oceanus-1, 2023_02_02__20_52_15, Turn 11

Pup (14hp) walks over two mines (24 dmg), dies and gets revived

**Comment by Zheneq** [comment]
Oceanus-3 - 0.7.2-beta - 2023_03_10__17_22_02  turn 13 
Tolren was not dead. He end turn with (-6)hp, and was still alive. Next turn at the start received +10 hp from the healing, and became "alive" (still got killed this turn). 
but he still could attack and survive (if had escapes)

**Comment by Zheneq** [comment]
Turn 6 tolren went to exactly 0 health in movement phase from a gremo mine then revived instantly the next turn with 1 health and the death was not counted. then the following turn Tolren went to 11 health. 
Oceanus-4 - 0.7.2-beta2 - 2023_03_22__02_20_11

**Comment by Zheneq** [comment]
It has to do something with healing over time

**Comment by Zheneq** [comment]
or not
2023_05_08__15_31_21 Cronus-2 0.7.3-beta
Turn 13, Lockwood loses last bits of hp on Gremo mine in chase phase, but stays alive only to die next turn having taken no damage.

**Comment by Zheneq** [comment]
Maybe it's resolved after normal movement but not resolved after chase?
Same goes for tech points.

Oceanus-1 - 0.7.3-beta - 2023_04_26__20_03_40 - Turn 7

```
[ActorData: Flight1ess#134, SpaceMarine, ActorIndex: 4, TeamA] [PlayerData: (3) Flight1ess#134] [Player: id=3, connectionId=4, accountId=1000001114803834] UnresolvedDamage.set 24, old: 0
[ActorData: SwitchySaint#626, Trickster, ActorIndex: 6, TeamB] [PlayerData: (5) SwitchySaint#626] [Player: id=5, connectionId=5, accountId=1000000770923326] UnresolvedTechPointGain.set 12, old: 0
Warning:<color=green>Effect</color>: <color=yellow>[ NO_NAME ] GUID 46</color> APPLIED
Caster: <color=yellow>[Trickster (SwitchySaint#626), 6]</color>, Target: <color=yellow>[SpaceMarine (Flight1ess#134), 4]</color>
Log:CheckForEndOfGame m_matchState: InMatch, m_skipEndOfGameCheck: False
Log:CheckForEndOfGame checking conditions
Log:CheckForEndOfGame turn 7/20, teamAPoints 0, teamBPoints 0, hasTeamAWon False, hasTeamBWon False
Log:GameFlowData::SetGameState BothTeams_Decision
Log:Game state: BothTeams_Decision
Log:Score: Team A: 0 Team B: 0
Log:Turn: 8
```

**Comment by Zheneq** [comment]
Probably fixed in 2ebd777e67920adac5aa64f94348246bc43fc70f, though I have my concerns regarding what triggered the bug. Not all interactions even in the end of chase movement phase failed to register the turn they happenned, maybe the bug happenned when one of the clients was lagging behind?
